File: src/main/java/ai/aimachineserver/infrastructure/UserRepositoryImpl.java
package ai.aimachineserver.infrastructure;

import ai.aimachineserver.domain.user.User;
import ai.aimachineserver.domain.user.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

interface UserCrudRepository extends CrudRepository<User, Long> {
    boolean existsByUsername(String username);

    User findByUsername(String username);

    User findUserById(Long id);
}

@Repository
class UserRepositoryImpl implements UserRepository {

    private final UserCrudRepository crudRepo;

    @Autowired
    UserRepositoryImpl(UserCrudRepository crudRepo) {
        this.crudRepo = crudRepo;
    }

    @Override
    public boolean existsByUsername(String username) {
        return crudRepo.existsByUsername(username);
    }

    @Override
    public User findByUsername(String username) {
        return crudRepo.findByUsername(username);
    }

    @Override
    public User findUserById(Long id) {
        return crudRepo.findUserById(id);
    }

    @Override
    public User save(User user) {
        return crudRepo.save(user);
    }
}

File: src/main/kotlin/ai/aimachineserver/infrastructure/UserRepositoryImpl.kt
package ai.aimachineserver.infrastructure

import ai.aimachineserver.domain.user.User
import ai.aimachineserver.domain.user.UserRepository
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository

interface UserCrudRepository : CrudRepository<User, Long> {
    fun existsByUsername(username: String): Boolean
    fun findByUsername(username: String): User?
    fun findUserById(id: Long): User?
}

@Repository
class UserRepositoryImpl(
    private val crudRepo: UserCrudRepository
) : UserRepository {
    override fun existsByUsername(username: String): Boolean = crudRepo.existsByUsername(username)
    override fun findByUsername(username: String): User? = crudRepo.findByUsername(username)
    override fun findUserById(id: Long): User? = crudRepo.findUserById(id)
    override fun save(user: User): User = crudRepo.save(user)
}